compiler/noirc_frontend/src/hir/resolution/import.rs

```diff
@@ -149,24 +155,29 @@ fn resolve_path_to_ns(
                 allow_contracts,
             )
         }
-        crate::ast::PathKind::Dep => resolve_external_dep(
-            def_map,
-            import_directive,
-            def_maps,
-            allow_contracts,
-            importing_crate,
-        ),
-        crate::ast::PathKind::Plain => {
-            // Plain paths are only used to import children modules. It's possible to allow import of external deps, but maybe this distinction is better?
-            // In Rust they can also point to external Dependencies, if no children can be found with the specified name
-            resolve_name_in_module(
+        crate::ast::PathKind::Plain | crate::ast::PathKind::Dep => {
+            let result = resolve_name_in_module(
                 crate_id,
                 importing_crate,
                 import_path,
                 import_directive.module_id,
                 def_maps,
                 allow_contracts,
-            )
+            );
+
+            // Attempt to resolve externally when unresolved
+            if let Err(ref result_err) = result {
+                if result_err.is_unresolved() {
+                    return resolve_external_dep(
+                        def_map,
+                        import_directive,
+                        def_maps,
+                        allow_contracts,
+                        importing_crate,
+                    );
+                }
+            }
```

Comment:
This also appears to make error messages worse for unresolved local paths. e.g. if I import `foo::bar::baz` but there's no definition for `baz` in the `bar` module then I'm pretty sure that this code will throw an error stating that it can't find `foo` (as it will be looking for a `foo` dependency in the fallback case.